### PR TITLE
Fix S2360 FP: Extend CallerInfoAttributes with CallerArgumentExpressionAttribute

### LINKED ISSUE #5925: Fix S2360 FP: Extend CallerInfoAttributes with CallerArgumentExpressionAttribute
### Description

C# 10 introduced [`CallerArgumentExpressionAttribute `](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.callerargumentexpressionattribute?view=net-6.0). It should be added to

https://github.com/SonarSource/sonar-dotnet/blob/74b01e58199b5a300d48fdb4fb97bf60260ab818/analyzers/src/SonarAnalyzer.Common/Helpers/KnownType.cs#L541-L545

Affected rules: 
* S2360
* S3236
* S3343

Affected RSpecs
* S3236
* S3343 

Reported by https://community.sonarsource.com/t/68752

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**
Fixes #5925

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.CSharp/Rules/ArgumentSpecifiedForCallerInfoParameter.cs [hunk 1]
```diff
@@ -34,55 +34,27 @@ public sealed class ArgumentSpecifiedForCallerInfoParameter : SonarDiagnosticAna
...
-                        var argument = argumentMapping.Node;
-
-                        var callerInfoAttributeDataOnCall = GetCallerInfoAttribute(parameter);
-                        if (callerInfoAttributeDataOnCall == null)
-                        {
-                            continue;
-                        }
-
-                        if (c.SemanticModel.GetSymbolInfo(argument.Expression).Symbol is IParameterSymbol symbolForArgument &&
-                            Equals(callerInfoAttributeDataOnCall.AttributeClass, GetCallerInfoAttribute(symbolForArgument)?.AttributeClass))
```
Comment (martin-strecker-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/ArgumentSpecifiedForCallerInfoParameter.cs:74]: The 
`Equals(callerInfoAttributeDataOnCall.AttributeClass, GetCallerInfoAttribute(symbolForArgument)?.AttributeClass)` part doesn't make sense IMO. The same test was done in line 67 before.

Comment (martin-strecker-sonarsource) [reply analyzers/src/SonarAnalyzer.CSharp/Rules/ArgumentSpecifiedForCallerInfoParameter.cs:74]: The `c.SemanticModel.GetSymbolInfo(argument.Expression).Symbol is IParameterSymbol symbolForArgument` is also useless. The `x.Symbol` in argumentMappings refer to this parameter already.

File: analyzers/src/SonarAnalyzer.CSharp/Rules/ArgumentSpecifiedForCallerInfoParameter.cs [hunk 2]
```diff
@@ -34,55 +34,27 @@ public sealed class ArgumentSpecifiedForCallerInfoParameter : SonarDiagnosticAna
         internal const string DiagnosticId = "S3236";
```
Comment (costin-zaharia-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/ArgumentSpecifiedForCallerInfoParameter.cs:34]: clean as you code: can be private

File: analyzers/src/SonarAnalyzer.CSharp/Rules/ArgumentSpecifiedForCallerInfoParameter.cs (hunk 1 above)
Comment (costin-zaharia-sonarsource) [reply analyzers/src/SonarAnalyzer.CSharp/Rules/ArgumentSpecifiedForCallerInfoParameter.cs:74]: Nice rewrite.

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM!

**Comment by martin-strecker-sonarsource** [comment]
This PR is an FP fix. Should it be milestoned? If so, is the issue or the PR or both milestoned?

**Comment by costin-zaharia-sonarsource** [comment]
> This PR is an FP fix. Should it be milestoned? If so, is the issue or the PR or both milestoned?

It should. We add all the issues that are directly impacting our users. If there is an issue we add the milestone to the issue, if not we add it to the PR.

**Pull request merged**